src/include/miopen/generic_search.hpp

```diff
@@ -338,143 +338,150 @@ auto GenericSearch(const Solver s, const Context& context, const AnyInvokeParams
                                << (useSpare ? " (spare)" : "")
                                << "...");
 
-    bool is_passed   = false; // left false only if all iterations failed.
-    float best_time  = std::numeric_limits<float>::max();
-    size_t n_failed  = 0;
-    size_t n_current = 0;
-    size_t n_best    = 0;
+    bool is_passed  = false; // left false only if all iterations failed.
+    float best_time = std::numeric_limits<float>::max();
+    size_t n_failed = 0;
+    size_t n_best   = 0;
     HeartBeat<PerformanceConfig> heartbeat;
     heartbeat.Start();
 
-    const char* const c_and_r = miopen::GetStringEnv(MIOPEN_COMPILE_AND_RUN{});
-    std::string compile_and_run;
-    if(c_and_r != nullptr && strlen(c_and_r) > 0)
+// PrecompileKernels call saves to binary_cache, this needs to be escaped if KERN_CACHE is not on.
+#if MIOPEN_ENABLE_SQLITE_KERN_CACHE
+    std::vector<KernelInfo> kernels;
+    for(const auto& current_config : all_configs)
     {
-        compile_and_run = c_and_r;
+        ConvSolution current_solution = s.GetSolution(context, current_config, true);
+        for(auto&& kernel : current_solution.construction_params)
+        {
+            if(profile_h.HasProgram(kernel.kernel_file, kernel.comp_options))
+                continue;
+            kernels.push_back(kernel);
+        }
     }
+    std::ignore = PrecompileKernels(profile_h, kernels);
+#endif
 
-    for(const auto& current_config : all_configs)
+    if(IsEnabled(MIOPEN_DEBUG_COMPILE_ONLY{}))
```

Comment:
:bug: Auto-tuning is not functional after this change. I am curious how this PR was tested.